### PR TITLE
Add ApiLimit and ApiBlock

### PR DESCRIPTION
**Pull request by TaiWilkin**
## Overview

Adds models to support metering API access. The ApiLimit describes the
number of Api requests contributors can make monthly. The ApiBlock
records times when the allowed amount has been exceeded and provides
a location for recording offered grace periods.

Connects #1138 

## Demo

ApiBlock
Screen Shot 2020-10-19 at 12 32 27 PM
ApiLimit
Screen Shot 2020-10-19 at 12 33 08 PM

## Testing Instructions

* Checkout this branch
* Run `./scripts/update`
* Use `./scripts/manage shell_plus` to create a new api limit and api block and to confirm the block was added
```
ApiLimit.objects.create(contributor_id=2, monthly_limit=50)
block = ApiBlock.objects.create(contributor_id=2, limit=50, actual=100, grace_limit=100, grace_reason='testing123',until='2020-10-30')
block.__dict__
```
* Navigate to [http://localhost:8081/admin/api/apilimit](http://localhost:8081/admin/api/apilimit) and confirm the limit exists
* Select the new limit and confirm that the monthly_limit can be updated but the contributor is readonly.

## Checklist

- [x] `fixup!` commits have been squashed
- [x] CI passes after rebase
- [x] CHANGELOG.md updated with summary of features or fixes, following [Keep a Changelog](https://keepachangelog.com/en/1.0.0/) guidelines

### CODE REVIEW

File: src/django/api/models.py [hunk 1]
```diff
@@ -1742,3 +1742,74 @@ class FacilityLocation(models.Model):
     )
     created_at = models.DateTimeField(auto_now_add=True)
     updated_at = models.DateTimeField(auto_now=True)
+
+
+class ApiLimit(models.Model):
+    """
+    Stores the number of requests a Contributor can make monthly.
+    """
+    contributor = models.ForeignKey(
```
Comment (jwalgran) [inline src/django/api/models.py:1751]: In our design notes we decided this should be a unique FK, but there is a system check warning suggesting that we change it to a `OneToOneField`. Consider making the suggested change. We already have a few of these, `Contributor.admin` for example

https://github.com/open-apparel-registry/open-apparel-registry/blob/6e2d71b190260edc975509611e86d0ed104627cf/src/django/api/models.py#L120-L124

If we to change the field type, we may need to regenerate the migration.

**Review by jwalgran — COMMENTED** [review]
Nice work adding comments and well worded field summaries. I have made a few inline suggestions.

In addition, I have been getting in the habit of adding `__str__` to all the models to make shell debugging easier and automatically have more information at a glance in the Django admin

Screen Shot 2020-10-20 at 9 17 22 AM


Screen Shot 2020-10-20 at 9 42 28 AM

Here is a suggested implementation

```python
    def __str__(self):
        return 'ApiLimit {id} - Contributor {contributor_id} limit {monthly_limit}'.format(**self.__dict__)
```

```python
    def __str__(self):
        return 'ApiBlock {id} - Contributor {contributor_id} until {until}'.format(**self.__dict__)

```

File: src/django/api/models.py [hunk 2]
```diff
@@ -1742,3 +1742,74 @@ class FacilityLocation(models.Model):
...
+        blank=False,
+        help_text='The limit value that was exceeded.')
+    actual = models.PositiveIntegerField(
+        null=False,
+        blank=False,
+        help_text='The count that exceeded the limit.')
+    grace_limit = models.PositiveIntegerField(
+        null=False,
+        blank=False,
+        help_text='An ad-hoc increase in the limit.')
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/models.py]: This was an error. Thank you for catching it! 👍

File: src/django/api/models.py [hunk 3]
```diff
@@ -1742,3 +1742,74 @@ class FacilityLocation(models.Model):
...
+    grace_created_by = models.ForeignKey(
+        'User',
+        null=True,
+        on_delete=models.SET_NULL,
+        help_text='The person who set the grace_limit.')
+    grace_reason = models.CharField(
```
[the remark this answers is not in the document]
Comment (TaiWilkin) [reply src/django/api/models.py]: Good call. I've updated this in fb3fde8

File: src/django/api/models.py (hunk 1 above)
Comment (TaiWilkin) [reply src/django/api/models.py]: Fixed in fb3fde8

**Comment by TaiWilkin** [comment]
@jwalgran I've updated the models and added __str__, and regenerated the migration. Be sure to update when you pull the code.

**Review by jwalgran — APPROVED** [review]
Changes look good. Thanks.

**Pull request merged**